integrations/maven-bloop/src/test/scala/bloop/integrations/maven/MavenConfigGenerationSuite.scala

```diff
@@ -0,0 +1,148 @@
+package bloop.integrations.maven
+
+import bloop.config.utils.BaseConfigSuite
+import java.io.BufferedReader
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+import java.io.InputStreamReader
+import java.io.File
+import org.junit.Assert._
+import org.junit.Test
+import org.apache.maven.shared.invoker.DefaultInvocationRequest
+import org.apache.maven.shared.invoker.DefaultInvoker
+import org.apache.maven.shared.invoker.InvocationResult
+import scala.jdk.CollectionConverters._
+import scala.util.control.NonFatal
+import scala.util.Try
+import scala.sys.process.ProcessLogger
+import bloop.config.Config
+import bloop.config.Tag
+
+class MavenConfigGenerationSuite extends BaseConfigSuite {
+
+  @Test
+  def basicScala3() = {
+    check("basic_scala3/pom.xml", projectName = "basic_scala3") { (configFile, projectName) =>
+      assert(configFile.project.`scala`.isDefined)
+      assertEquals("3.0.0", configFile.project.`scala`.get.version)
+      assertEquals("org.scala-lang", configFile.project.`scala`.get.organization)
+      assert(configFile.project.`scala`.get.jars.exists(_.toString.contains("scala3-compiler_3")))
+      assert(hasCompileClasspathEntryName(configFile, "scala3-library_3"))
+      assert(hasCompileClasspathEntryName(configFile, "scala-library"))
+
+      val idxDottyLib = idxOfClasspathEntryName(configFile, "scala3-library_3")
+      val idxScalaLib = idxOfClasspathEntryName(configFile, "scala-library")
+
+      assert(idxDottyLib < idxScalaLib)
+
+      assert(hasTag(configFile, Tag.Library))
+
+      assertNoConfigsHaveAnyJars(List(configFile), List(s"$projectName", s"$projectName-test"))
+      assertAllConfigsMatchJarNames(List(configFile), List("scala3-library_3"))
+    }
+  }
+
+  @Test
+  def basicScala() = {
+    check("basic_scala/pom.xml", projectName = "basic_scala") { (configFile, projectName) =>
+      assert(configFile.project.`scala`.isDefined)
+      assertEquals("2.13.6", configFile.project.`scala`.get.version)
+      assertEquals("org.scala-lang", configFile.project.`scala`.get.organization)
+      assert(
+        !configFile.project.`scala`.get.jars.exists(_.toString.contains("scala3-compiler_3")),
+        "No Scala 3 jar should be present."
+      )
+      assert(!hasCompileClasspathEntryName(configFile, "scala3-library_3"))
+      assert(hasCompileClasspathEntryName(configFile, "scala-library"))
+
+      assert(hasTag(configFile, Tag.Library))
+
+      assertNoConfigsHaveAnyJars(List(configFile), List(s"$projectName", s"$projectName-test"))
+      assertAllConfigsMatchJarNames(List(configFile), List("scala-library", "munit"))
+    }
+  }
+
+  private def check(testProject: String, projectName: String)(
+      checking: (Config.File, String) => Unit
+  ): Unit = {
+    val tempDir = Files.createTempDirectory("mavenBloop")
```

Comment:
In order to avoid some additional dependencies on Maven testers (which are not so simple to add in sbt) the tests are done this:
- publishM2 is run before the tests
- maven process is executed using the maven wrapper
- configuration is read and checked